eth/chains/base.py

```diff
@@ -277,6 +280,10 @@ def import_block(self,
     #
     # Validation API
     #
+    @abstractmethod
+    def validate_receipt(self, block_number: BlockNumber, receipt: Receipt) -> None:
```

Comment:
The arguments felt in the wrong order to me. I checked some other methods that are basically VM passthroughs, and they flip the order of arguments, so it would be nice to keep consistency if possible. Also, most other places are using the header instead of the block_number, which looks like it would be okay.

I'm looking at:
- `get_transaction_result()`
- `estimate_gas()`
- `build_block_with_transactions()`

So I think an ideal option here is: 

```py
    def validate_receipt(self, receipt: Receipt, at_header: BlockHeader) -> None:
```

Reply:
I originally chose `block_number` because I wasn't sure I'd have the full header, but it looks like the header is available everywhere I need it.